tests/phpunit/tests/ajax/CropImage.php

```diff
@@ -0,0 +1,158 @@
+
+		wp_delete_attachment( $attachment->ID, true );
+		wp_delete_attachment( $cropped_attachment->ID, true );
+	}
+
+	/**
+	 * Tests that attachment properties are not auto-generated if they are not defined for the original image.
+	 *
+	 * @ticket 37750
+	 *
+	 * @covers ::wp_ajax_crop_image
+	 */
+	public function test_it_doesnt_generate_new_metadata_if_metadata_is_empty() {
+		// Become an administrator.
+		$this->_setRole( 'administrator' );
+
+		$attachment = $this->create_attachment( false );
+		$this->prepare_post( $attachment );
+
+		// Make the request.
+		try {
+			$this->_handleAjax( 'crop-image' );
+		} catch ( WPAjaxDieContinueException $e ) {
+		}
+
+		$response = json_decode( $this->_last_response, true );
+		$this->validate_response( $response );
+
+		$cropped_attachment = get_post( $response['data']['id'] );
+		$this->assertInstanceOf( WP_Post::class, $cropped_attachment, 'get_post function must return an instance of WP_Post class' );
+		$this->assertEmpty( $attachment->post_title, 'post_title value must be empty for testing purposes' );
+		$this->assertNotEmpty( $cropped_attachment->post_title, 'post_title value must be auto-generated if it\'s empty in the original attachment' );
+		$this->assertStringStartsWith( 'http', $cropped_attachment->post_content, 'post_content value should contain an URL if it\'s empty in the original attachment' );
+		$this->assertEmpty( $cropped_attachment->post_excerpt, 'post_excerpt value must be empty if it\'s empty in the original attachment' );
+		$this->assertEmpty( $cropped_attachment->_wp_attachment_image_alt, '_wp_attachment_image_alt value must be empty if it\'s empty in the original attachment' );
+
+		wp_delete_attachment( $attachment->ID, true );
+		wp_delete_attachment( $cropped_attachment->ID, true );
+	}
+
+	/**
+	 * Creates an attachment.
+	 *
+	 * @return WP_Post
+	 */
+	private function create_attachment( $with_metadata = true ) {
+		$uniq_id = uniqid( 'crop-image-ajax-action-test-' );
+		$object  = array(
+			'post_title'     => $with_metadata ? 'Title ' . $uniq_id : '',
+			'post_content'   => $with_metadata ? 'Description ' . $uniq_id : '',
+			'post_mime_type' => 'image/jpg',
+			'guid'           => 'http://localhost/foo.jpg',
+			'context'        => 'custom-logo',
+			'post_excerpt'   => $with_metadata ? 'Caption ' . $uniq_id : '',
+		);
+
+		$test_file        = DIR_TESTDATA . '/images/test-image.jpg';
+		$upload_directory = wp_upload_dir();
+		$uploaded_file    = $upload_directory['path'] . '/' . $uniq_id . '.jpg';
+		$filesystem       = new WP_Filesystem_Direct( true );
+		$filesystem->copy( $test_file, $uploaded_file );
+		$attachment_id = wp_insert_attachment( $object, $uploaded_file );
+
+		if ( $with_metadata ) {
+			update_post_meta( $attachment_id, '_wp_attachment_image_alt', wp_slash( 'Alt ' . $uniq_id ) );
+		}
+
+		return get_post( $attachment_id );
+	}
```

Comment:
To avoid duplication and inconsistencies, I'd suggest instead to use `$this->_make_attachment()` helper method (which is available through inheritance coming from `WP_UnitTestCase_Base`). If `$with_metadata` is `true`, then update the post and its metadata. Doing this ensures the logic needed to create an attachment is centralized in one location.

Reply:
> I'd suggest instead to use $this->_make_attachment() helper method 
That's good to know. Fixed in a39a55141d5da872225890a2054722f567a657de